### PR TITLE
Implement Gunit-style refactoring

### PR DESCRIPTION
**Pull request by e0426051**
Assertions does not seem to work here even after I enabled it though

### CODE REVIEW

**Comment by keke101** [comment]
Have you tried enabling the assertions? [Link](https://stackoverflow.com/questions/18168257/where-to-add-compiler-options-like-ea-in-intellij-idea)

File: src/main/java/seedu/duke/command/task/ViewTaskCommand.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+        try {
+            Project proj = projectListManager.getProject();
+            if (parameters.isEmpty()) {
+                Ui.showError("Missing parameters.");
+            }
+            Ui.showToUserLn("The details of the tasks are as follows: ");
+            for (int i = 0; i < parameters.size(); i++) {
+                Task task;
+                try {
```
Comment (keke101) [inline src/main/java/seedu/duke/command/task/ViewTaskCommand.java]: I think no need to nest another `try-catch`, you can just put another catch for `NumberFormatException` in the other `try-catch`.

File: src/main/java/seedu/duke/model/task/ProjectBacklog.java [hunk 2]
```diff
@@ -47,6 +55,10 @@ public void addTask(String title, String description, String priority) {
         backlogTasks.add(new Task(newTaskId, title, description, priority));
     }
 
+    public int idSize() {
```
Comment (keke101) [inline src/main/java/seedu/duke/model/task/ProjectBacklog.java]: Got a getter for `nextId` already, at line 37

**Review by keke101 — CHANGES_REQUESTED** [review]

File: src/main/java/seedu/duke/model/task/ProjectBacklog.java (hunk 2 above)
Comment (gmit22) [reply src/main/java/seedu/duke/model/task/ProjectBacklog.java]: Think remove it then

**Review by gmit22 — APPROVED** [review]

**Comment by e0426051** [comment]
I have merged the try-catch block and removed the duplicate function from the classes.


> Have you tried enabling the assertions? [Link](https://stackoverflow.com/questions/18168257/where-to-add-compiler-options-like-ea-in-intellij-idea)

I enabled it from settings, and also ran someone's test code which show the thing is enabled. But then like if I run "project /view" from the start it still exits with error even though got assert, then I can only fix it with try-catch block. Also if I replace the "no project" errors from try-catch to assertion then the error message prints but the task is still added. I'm not sure if this is normal behavior. Maybe assertions cannot replace try-catch?

**Comment by chocomango** [comment]
> 
> 
> I have merged the try-catch block and removed the duplicate function from the classes.
> 
> > Have you tried enabling the assertions? [Link](https://stackoverflow.com/questions/18168257/where-to-add-compiler-options-like-ea-in-intellij-idea)
> 
> I enabled it from settings, and also ran someone's test code which show the thing is enabled. But then like if I run "project /view" from the start it still exits with error even though got assert, then I can only fix it with try-catch block. Also if I replace the "no project" errors from try-catch to assertion then the error message prints but the task is still added. I'm not sure if this is normal behavior. Maybe assertions cannot replace try-catch?

The error was because there is no check whether there is a project before executing _ViewProjectCommand_. The error has already been fixed by @gmit22 in his latest PR.

**Comment by chocomango** [comment]
I have merged his PR, so you can do a pull on the repo and see if the error persist.

**Comment by keke101** [comment]
You are talking about the Unit Test assertions is it? Those assertions are different from the normal assertions. Normal assertions are placed in the code using the `assert` keyword, therefore the link I have provided only works for normal assertions. Anyway both have different purpose, so yeah.

**Review by keke101 — APPROVED** [review]

**Comment by keke101** [comment]
I think we can merge this one, then you create another PR request for the JUnit.

**Comment by e0426051** [comment]
> > I have merged the try-catch block and removed the duplicate function from the classes.
> > > Have you tried enabling the assertions? [Link](https://stackoverflow.com/questions/18168257/where-to-add-compiler-options-like-ea-in-intellij-idea)
> > 
> > 
> > I enabled it from settings, and also ran someone's test code which show the thing is enabled. But then like if I run "project /view" from the start it still exits with error even though got assert, then I can only fix it with try-catch block. Also if I replace the "no project" errors from try-catch to assertion then the error message prints but the task is still added. I'm not sure if this is normal behavior. Maybe assertions cannot replace try-catch?
> 
> The error was because there is no check whether there is a project before executing _ViewProjectCommand_. The error has already been fixed by @gmit22 in his latest PR.


Just checked again, the code is working as expected now.


> You are talking about the Unit Test assertions is it? Those assertions are different from the normal assertions. Normal assertions are placed in the code using the `assert` keyword, therefore the link I have provided only works for normal assertions. Anyway both have different purpose, so yeah.

I think the problem is I did not pull Gunit's latest code. Now everything looks good.

I made more changes regarding some commands without dashes, so maybe can review one more time before merge.

**Comment by gmit22** [comment]
Resolve the conflicts, youd have to do it using cli in your own repo @e0426051

**Pull request merged**